### PR TITLE
Member Edit Error Handling

### PR DESCRIPTION
**Pull request by eparubrub**
- Adds API error handling to Member Edit
- Checking email before sending for users (timeout of 5 seconds if no associated email on record)
- Adds link to email sent page in case users aren't receiving it

### CODE REVIEW

File: pages/edit/index.tsx [hunk 1]
```diff
@@ -116,6 +116,19 @@ function EditForm({ baseUrl }) {
...
+            ) {
+              return new Promise((resolve) => {
+                setTimeout(() => {
+                  setPageState(PageState.Error);
+                  setError(
+                    `Looks like the email ${email} is not associated with a ` +
+                      `Hawaiians in Tech account`,
+                  );
+                  resolve(response);
+                }, 5000); // 5-second delay if the email is not found
```
Comment (tellaho) [inline pages/edit/index.tsx]: 5-seconds feels like an eternity, especially for a superficial return. I get we're trying to slow attempts down though.

As a compromise, thoughts on making it `2000`?

File: pages/edit/index.tsx [hunk 2]
```diff
@@ -116,6 +116,19 @@ function EditForm({ baseUrl }) {
...
                   "cloudflare check. Please try again.",
               );
+            } else if (
+              json.message.includes(`Member with email ${email} not found`)
+            ) {
+              return new Promise((resolve) => {
+                setTimeout(() => {
+                  setPageState(PageState.Error);
+                  setError(
+                    `Looks like the email ${email} is not associated with a ` +
```
Comment (tellaho) [inline pages/edit/index.tsx]: Small tweak to entertain me: `We don't recognize the email ${email}. Did you sign up with another?`

image

**Review by tellaho — APPROVED** [review]
Gave it a go, works as expected. Appreciate the look at this, man.

Couple small adjustments, between shortening the timeout and tweaking the copy. Other than that, good to go (ps opting for approving).

File: pages/edit/index.tsx (hunk 1 above)
Comment (eparubrub) [reply pages/edit/index.tsx]: updated to 2000 in commit [457c7a6](https://github.com/hawaiians/hawaiiansintech/pull/36/commits/457c7a64374b991c99047eeee5e38b4e052388fe)

File: pages/edit/index.tsx (hunk 2 above)
Comment (eparubrub) [reply pages/edit/index.tsx]: updated with commit [457c7a6](https://github.com/hawaiians/hawaiiansintech/pull/36/commits/457c7a64374b991c99047eeee5e38b4e052388fe)

**Pull request merged**